Local build Failed on MacOS 13.2 Beta x86_64

<!--## Prerequisites

- First, many thanks for taking part in the community. We really appreciate that.
- We realize there is a lot of information requested here. We ask only that you do your best to provide as much information as possible so we can better help you.
- Support questions are better asked in one of the following locations:
    - [Solidity chat](https://gitter.im/ethereum/solidity)
    - [Stack Overflow](https://ethereum.stackexchange.com/)
- Ensure the issue isn't already reported.
- The issue should be reproducible with the latest solidity version; however, this isn't a hard requirement and being reproducible with an older version is sufficient.

*Delete the above section and the instructions in the sections below before submitting*
-->

## Description

Attempt to compile Solidity locally but failed. The error log shows: "solidity/libsolidity/lsp/DocumentHoverHandler.cpp:66:27: error: chosen constructor is explicit in copy-initialization
        MarkdownBuilder markdown{};"

## Environment

- Operating system: MacOS 13.2 Beta
- CPU: Intel x86_64 
- Clang: Apple clang version 11.0.0 (clang-1100.0.33.17) Target: x86_64-apple-darwin22.3.0
- codebase version: https://github.com/ethereum/solidity/commit/2e2210222c1d68f8dd5710cd4911bf34f434571b

## Steps to Reproduce

- run ./scripts/build.sh

Errors:


```
[ 54%] Building CXX object libsolidity/CMakeFiles/solidity.dir/lsp/DocumentHoverHandler.cpp.o
[ 55%] Building CXX object libsolidity/CMakeFiles/solidity.dir/lsp/FileRepository.cpp.o
solidity/libsolidity/lsp/DocumentHoverHandler.cpp:66:27: error: chosen constructor is explicit in copy-initialization
        MarkdownBuilder markdown{};
solidity/libsolidity/lsp/DocumentHoverHandler.cpp:38:15: note: in implicit initialization of field 'result' with omitted initializer
        stringstream result;
                     ^
1 error generated.
1 error generated.
make[2]: *** [libsolidity/CMakeFiles/solidity.dir/lsp/DocumentHoverHandler.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [libsolidity/CMakeFiles/solidity.dir/all] Error 2
make: *** [all] Error 2
```


<!--
Please provide a *minimal* source code example to trigger the bug you have found.
Please also mention any command line flags that are necessary for triggering the bug.
Provide as much information as necessary to reproduce the bug.



```solidity
// Some *minimal* Solidity source code to reproduce the bug.
// ...
```
-->

--- a/projects/soliditylang.org/clang-error.diff
+++ b/projects/soliditylang.org/clang-error.diff
@@ -1,0 +1,12 @@
+diff --color=auto -ru a/libsolidity/lsp/DocumentHoverHandler.cpp b/libsolidity/lsp/DocumentHoverHandler.cpp
+--- a/libsolidity/lsp/DocumentHoverHandler.cpp	2023-02-01 17:33:07
++++ b/libsolidity/lsp/DocumentHoverHandler.cpp	2023-02-01 17:32:26
+@@ -63,7 +63,7 @@
+ 	auto const [sourceUnitName, lineColumn] = HandlerBase(*this).extractSourceUnitNameAndLineColumn(_args);
+ 	auto const [sourceNode, sourceOffset] = m_server.astNodeAndOffsetAtSourceLocation(sourceUnitName, lineColumn);
+ 
+-	MarkdownBuilder markdown{};
++ MarkdownBuilder markdown = {stringstream(ios_base::in)};
+ 	auto rangeToHighlight = toRange(sourceNode->location());
+ 
+ 	// Try getting the type definition of the underlying AST node, if available.

--- a/projects/soliditylang.org/package.yml
+++ b/projects/soliditylang.org/package.yml
@@ -24,8 +24,15 @@ build:
     freedesktop.org/pkg-config: ^0.29
     git-scm.org: 2
     crates.io/semverator: '*'
+    gnu.org/patch: '*'
   working-directory: build
   script: |
+    if semverator satisfies ^0.8.18 {{version}}; then
+      # New libsolidity/lsb/DocumentHoverHandler.cpp doesn't like some versions of clang
+      # (default argument issue)
+      patch -p1 -d.. <../props/clang-error.diff
+    fi
+
     cmake .. $ARGS
     make --jobs {{hw.concurrency}}
     make install

--- a/projects/soliditylang.org/package.yml
+++ b/projects/soliditylang.org/package.yml
@@ -23,12 +23,16 @@ build:
     cmake.org: 3
     freedesktop.org/pkg-config: ^0.29
     git-scm.org: 2
+    crates.io/semverator: '*'
   working-directory: build
   script: |
     cmake .. $ARGS
     make --jobs {{hw.concurrency}}
     make install
-    rm {{prefix}}/bin/solidity-upgrade
+
+    if semverator lt {{version}} 0.8.18; then
+      rm {{prefix}}/bin/solidity-upgrade
+    fi
   test:
     make test
   env:

